tests/Type/data/arrayable.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Arrayable;
+
+use Illuminate\Contracts\Support\Arrayable;
+use function PHPStan\Testing\assertType;
+
+/**
+ * @implements Arrayable<string, int>
+ */
+class Foo implements Arrayable
+{
+    public function toArray()
+    {
+
+    }
+}
+
+function test(): void
+{
+    $foo = new Foo;
+    assertType('array<string, int>', $foo->toArray());
+}
```

Comment:
This is good, but here's how we generally write the tests:

```suggestion
function test(Foo $foo): void
{
    assertType('array<string, int>', $foo->toArray());
}

/** @implements Arrayable<string, int>*/
class Foo implements Arrayable
{
    public function toArray()
    {
        return [];
    }
}
```


However, it's up to you if you want to make the changes.

Reply:
Thanks. Any reason we leave the return type declaration from toArray?

Reply:
You can add the php return type, but there's no reason to duplicate the `@return` because it's already defined on the parent:

```php
{
    public function toArray(): array
    {
        return [];
    }
```


You can also use `/** @inheritDoc */` or `#[Override]` (preferred, but PHP >=8.4) if that helps you remember, but it's not necessary:

```php
{
    /** @inheritDoc */
    public function toArray(): array
    {
        return [];
    }
```

Reply:
I guess my personal attitude has always been that static analysis (or code analysis in this case) should build on top of the native type system - rather than replace it entirely.

Fully appreciate this is a rather simple example - just curious if this is the approach you (given your wider experience) would take for other situations too, and if so why?